### PR TITLE
Add message methods

### PR DESCRIPTION
**Pull request by grgsrs**
This PR addresses issue #16 and adds support for the following message methods:

- `Clear`
- `ClearField`
- `CopyFrom`
- `HasField`
- `IsInitialized`
- `MergeFrom`
- `MergeFromString`
- `SetInParent`
- `SerializePartialToString`

I have removed message validation from `ParseFromString` as it is not performed in protobuf's `ParseFromString` method. Also, message constructors and the `TypedList.add` method now support keyword arguments as protobuf's equivalents do.

Finally, this PR should also fix issue #8.

### CODE REVIEW

File: pyrobuf/protobuf/templates/proto_pyx.tmpl [hunk 1]
```diff
@@ -42,40 +80,42 @@ cdef class {{ name }}:
             components.append('}')
         return '\n'.join(components)
 
-    cpdef reset(self):
+    cpdef void reset(self):
```
Comment (tburmeister) [inline pyrobuf/protobuf/templates/proto_pyx.tmpl]: Does it make sense to combine `reset` and `Clear`? If not, should `reset` be made a "private" method?

**Comment by tburmeister** [comment]
This is great! Will let @mmoresco make the final call on merging.

**Comment by mmoresco** [comment]
Looks good to me! This is really exciting, awesome work.

**Comment by mmoresco** [comment]
Available as version 0.5.5 in pypi!

**Pull request merged**